src/components/icon/icon.component.js

```diff
@@ -51,26 +51,29 @@ const Icon = React.forwardRef(
       }
     };
 
+    const styleProps = {
+      bg,
+      bgTheme,
+      bgSize,
+      bgShape,
+      color,
+      disabled,
+      fontSize,
+      iconColor,
+      mr,
+      ml,
+      tabIndex,
+      type: iconType(),
+    };
+
     const icon = (
       <StyledIcon
         ref={ref}
-        bgSize={bgSize}
-        bgShape={bgShape}
-        bgTheme={bgTheme}
-        fontSize={fontSize}
-        iconColor={iconColor}
-        disabled={disabled}
-        color={color}
-        bg={bg}
-        type={iconType()}
         key="icon"
         className={className || null}
         {...tagComponent("icon", rest)}
         data-element={iconType()}
-        mr={mr}
-        ml={ml}
-        tabIndex={tabIndex}
-        {...rest}
```

Comment:
I thought we decided we wanted our pattern to be always spread `...rest` on the root element of a component so that users can pass `data-element` etc to it?

Reply:
data-element is already passed using the `tagComponent` method. As for the other props - this is the task description: "Prevent unwanted props from being passed down to Icon".

Reply:
I think you'd removed `tagComponent` when I commented this but looks like it's back now so that'll do